pkg/resources/management.cattle.io/v3/project/quota_validate.go

```diff
@@ -0,0 +1,61 @@
+package project
+
+import (
+	mgmtv3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/wrangler/pkg/data/convert"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	quotav1 "k8s.io/apiserver/pkg/quota/v1"
+)
+
+// quotaFits checks whether the quota in the second argument is sufficient for the requested quota in the first argument.
+// If it is not sufficient, a list of the resources that exceed the allotment is returned.
+// The ResourceList to be checked can be compiled by passing a ResourceQuotaLimit to convertLimitToResourceList,
+// or a slice of ResourceQuotaLimit to convertLimitsToResourceList, before calling this function on the result.
+func quotaFits(resourceList corev1.ResourceList, projectLimit *mgmtv3.ResourceQuotaLimit) (bool, corev1.ResourceList, error) {
+	projectResourceList, err := convertLimitToResourceList(projectLimit)
+	if err != nil {
+		return false, nil, err
+	}
+
+	_, exceeded := quotav1.LessThanOrEqual(resourceList, projectResourceList)
+	// Include resources with negative values among exceeded resources.
+	exceeded = append(exceeded, quotav1.IsNegative(resourceList)...)
+	if len(exceeded) == 0 {
+		return true, nil, nil
+	}
+	failedHard := quotav1.Mask(resourceList, exceeded)
+	return false, failedHard, nil
+}
+
+// convertLimitToResourceList converts a management.cattle.io/v3 ResourceQuotaLimit object to a core/v1 ResourceList,
+// which can then be used to compare quotas.
+func convertLimitToResourceList(limit *mgmtv3.ResourceQuotaLimit) (corev1.ResourceList, error) {
+	toReturn := corev1.ResourceList{}
+	converted, err := convert.EncodeToMap(limit)
+	if err != nil {
+		return nil, err
+	}
+	for key, value := range converted {
+		q, err := resource.ParseQuantity(convert.ToString(value))
+		if err != nil {
+			return nil, err
+		}
+		toReturn[corev1.ResourceName(key)] = q
+	}
+	return toReturn, nil
+}
+
+// convertLimitsToResourceList converts a slice of management.cattle.io/v3 ResourceQuotaLimit objects to a single core/v1 ResourceList
+// by adding their results together. This can then be used to compare the total of all the quota limits to another limit.
+func convertLimitsToResourceList(limits []*mgmtv3.ResourceQuotaLimit) (corev1.ResourceList, error) {
```

Comment:
Looks like this function isn't being used. Is it here for future validations?

Reply:
Leftover from when I was validating the default for the existing namespaces, will remove, thanks for catching